ff/src/biginteger/mod.rs

```diff
@@ -421,6 +421,26 @@ impl<const N: usize> BigInteger for BigInt<N> {
         }
     }
 
+    #[inline]
+    fn mul(&mut self, other: &Self) {
+        if self.is_zero() || other.is_zero() {
+            *self = Self::zero();
+            return;
+        }
+
+        let mut res = Self::zero();
+        let mut carry = 0;
+
+        for i in 0..N {
+            for j in 0..(N - i) {
+                res.0[i + j] = mac_with_carry!(res.0[i + j], self.0[i], other.0[j], &mut carry);
+            }
+            carry = 0;
+        }
+
+        *self = res
+    }
+
```

Comment:
Let's to add a `mul_high` method that returns the high part of the result (i.e. the carry), and let's rename this method to `mul_low`, adjusting the documentation appropriately

Reply:
Oh and the `mul` method can then return both a tuple containing both the high and low parts. This is useful for some applications.

Reply:
Should be fixed now